### PR TITLE
return boolean from storage.delete

### PR DESCRIPTION
**Pull request by mehtaphysical**
Update `storage.delete(key)` so that it returns true if the key was present in storage before the delete. `storage_remove` returns this information: https://github.com/near/NEPs/blob/master/specs/RuntimeSpec/Components/BindingsSpec/TrieAPI.md#returns-2

This will make it easier to implement the `remove` method for #543 (see the [rust implementation](https://docs.rs/near-sdk/3.1.0/near_sdk/collections/struct.LazyOption.html#impl-1))

### CODE REVIEW

File: sdk-core/assembly/storage.ts [hunk 1]
```diff
@@ -125,11 +125,14 @@ export class Storage {
...
+   * @param key The unique identifier associated with a value in a key-value Store
+   * @returns True if the given key was present in storage
    */
-  static delete(key: string): void {
+  static delete(key: string): bool {
     let key_encoded = util.stringToBytes(key);
-    env.storage_remove(key_encoded.byteLength, key_encoded.dataStart, 0);
+    let status = env.storage_remove(key_encoded.byteLength, key_encoded.dataStart, 0);
+
+    return status == 1;
```
Comment (willemneal) [inline sdk-core/assembly/storage.ts:135]: Since `bool` is `1` or `0`, this is equivalent to `status == true`.
```suggestion
    return status;
```

**Review by willemneal — APPROVED** [review]
LGTM, just my one nit.

File: sdk-core/assembly/storage.ts [hunk 2]
```diff
@@ -125,11 +125,14 @@ export class Storage {
...
+   * @param key The unique identifier associated with a value in a key-value Store
+   * @returns True if the given key was present in storage
    */
-  static delete(key: string): void {
+  static delete(key: string): bool {
     let key_encoded = util.stringToBytes(key);
-    env.storage_remove(key_encoded.byteLength, key_encoded.dataStart, 0);
+    let status = env.storage_remove(key_encoded.byteLength, key_encoded.dataStart, 0);
+
+    return status;
```
Comment (willemneal) [inline sdk-core/assembly/storage.ts:135]:
```suggestion
    return <bool>status;
```
Ah it seems I was wrong since `u64`s aren't coerced into `bool`s.

Comment (mehtaphysical) [reply sdk-core/assembly/storage.ts]: Ah my bad

**Pull request merged**